### PR TITLE
Improve CCIP Contract Reader Querying usage

### PR DESCRIPTION
**Pull request by ilija42**
related chainlink-ccip PR https://github.com/smartcontractkit/chainlink-ccip/pull/267

Also relates to 
[CCIP-3866](https://smartcontract-it.atlassian.net/browse/CCIP-3866)
and
[CCIP-3865 ](https://smartcontract-it.atlassian.net/browse/CCIP-3865)

[CCIP-3866]: https://smartcontract-it.atlassian.net/browse/CCIP-3866?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by patrick-dowell — DISMISSED** [review]
I think the code itself is okay as is, but it's pretty hard to follow and seems like it'd be a little fragile if someone came in and tried to change things. 

I'd recommend more code comments to explain what the various methods are actually doing and, in addition, consider adding unit tests to the helper methods themselves to verify that they are calculating the indices properly. This would make it easier for someone to come in and modify things without breaking some of the high level tests and struggle to figure out what went wrong.

**Review by silaslenihan — DISMISSED** [review]
LGTM

**Review by Farber98 — DISMISSED** [review]

**Review by mateusz-sekara — DISMISSED** [review]

**Review by makramkd — DISMISSED** [review]

**Review by mateusz-sekara — APPROVED** [review]

**Review by silaslenihan — APPROVED** [review]

**Review by EasterTheBunny — APPROVED** [review]

**Review by jmank88 — APPROVED** [review]
